src/NATS.Client.Services/NatsSvcService.cs

```diff
@@ -0,0 +1,329 @@
+using System.Collections.Concurrent;
+using System.Text.Json.Nodes;
+using System.Threading.Channels;
+using Microsoft.Extensions.Logging;
+using NATS.Client.Core;
+using NATS.Client.Core.Internal;
+using NATS.Client.Services.Internal;
+using NATS.Client.Services.Models;
+
+namespace NATS.Client.Services;
+
+/// <summary>
+/// NATS service.
+/// </summary>
+public class NatsSvcService : IAsyncDisposable
```

Comment:
Maybe call it `NatsSvcServer` ?  Seems to be serving a service, and `SvcService` is a little confusing